### PR TITLE
Spike ins2

### PR DESCRIPTION
**Pull request by siebrenf**
**What problem is the PR solving / What's new?**
Extended PR to `spike ins`. In this version the custom regions are used everywhere. Prior only for indexing.

cleaner: 
  - the `assembly` wildcard.
  - consistent usage of the custom assembly

uglier: 
  - anything that used `sample['assembly'] == wilcards.assembly` is now `sample['assembly'] == ori_assembly(wilcards.assembly)`
  - several file paths build using `for sample in samples.index` (and similar): `{samples.loc[sample]['assembly']}{suffix}` (`{suffix}` added)

### CODE REVIEW

**Review by Maarten-vd-Sande — APPROVED** [review]
I just scrolled through :angel: 

Why two PRs?

**Comment by siebrenf** [comment]
The changes in `rule trackhub.smk` contain all the ugly code changes I had to do to make wildcard `assembly` fit both all assemblies and all custom assemblies. 

The function `ori_assembly` and `suffix` are defined in the configuration.

File: seq2science/rules/gene_counts.smk [hunk 1]
```diff
@@ -3,11 +3,11 @@ if config["quantifier"] == "salmon":
     def get_counts(wildcards):
         quant_dirs = []
         if config["quantifier"] in ["salmon", "star"]:
-            for sample in treps[treps["assembly"] == wildcards.assembly].index:
+            for sample in treps[treps["assembly"] == ori_assembly(wildcards.assembly)].index:
```
Comment (Maarten-vd-Sande) [inline seq2science/rules/gene_counts.smk:6]: what does ori_assembly stand for? Original assembly?

**Review by Maarten-vd-Sande — APPROVED** [review]

File: seq2science/rules/gene_counts.smk (hunk 1 above)
Comment (siebrenf) [reply seq2science/rules/gene_counts.smk:6]: yup!

**Pull request merged**